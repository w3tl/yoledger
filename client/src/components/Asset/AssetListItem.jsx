import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { assetPropType } from './propTypes';

function AssetListItem({ url, asset }) {
  const { id, name } = asset;
  return (
    <li key={name}>
      {url ? (
        <Link to={{ pathname: `${url}/view`, state: { id } }}>{name}</Link>
      ) : (
        name
      )}
    </li>
  );
}

AssetListItem.propTypes = {
  url: PropTypes.string,
  asset: assetPropType.isRequired,
};

AssetListItem.defaultProps = {
  url: null,
};

export default AssetListItem;
