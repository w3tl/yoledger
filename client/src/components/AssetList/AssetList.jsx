import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { accountPropType } from '../propTypes';
import withData from './AssetListHOC';

function AssetList({ assets, match }) {
  return (
    <ul>
      {assets.map(account => (
        <li key={account.name}>
          <Link to={`${match.url}/${account.id}`}>{account.name}</Link>
        </li>
      ))}
    </ul>
  );
}

AssetList.propTypes = {
  assets: PropTypes.arrayOf(accountPropType),
  match: PropTypes.shape({
    url: PropTypes.string,
  }),
};

AssetList.defaultProps = {
  match: {
    url: '',
  },
  assets: [],
};

export default withData(AssetList);
