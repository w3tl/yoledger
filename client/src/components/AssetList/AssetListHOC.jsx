import React from 'react';

const assets = [{
  id: '1',
  name: 'Bank Card',
}, {
  id: '2',
  name: 'Cash',
}, {
  id: '3',
  name: 'Bank Card 2',
}];

const AssetsWithData = Component => props => (
  <Component assets={assets} {...props} />
);

export default AssetsWithData;
