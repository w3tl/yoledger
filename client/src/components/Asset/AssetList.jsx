import React from 'react';
import PropTypes from 'prop-types';
import AssetListItem from './AssetListItem';
import { assetPropType } from './propTypes';

function AssetList({ assets, match }) {
  return (
    <ul>
      {assets.map(account => (
        <AssetListItem key={account.name} url={match.url} asset={account} />))}
    </ul>
  );
}

AssetList.propTypes = {
  assets: PropTypes.arrayOf(assetPropType),
  match: PropTypes.shape({
    url: PropTypes.string,
  }),
};

AssetList.defaultProps = {
  match: {
    url: '',
  },
  assets: [],
};

export default AssetList;
