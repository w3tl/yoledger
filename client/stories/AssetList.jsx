import React from 'react';
import { storiesOf } from '@storybook/react';

import AssetList from '../src/components/Asset/AssetList';

const assets = [{
  id: '1',
  name: 'Bank Card',
  balance: 0,
}, {
  id: '2',
  name: 'Cash',
  balance: 10,
}, {
  id: '3',
  name: 'Bank Card 2',
  balance: 150,
}];

storiesOf('Asset/List', module)
  .add('with data', () => <AssetList assets={assets} />)
  .add('empty list', () => <AssetList />);
