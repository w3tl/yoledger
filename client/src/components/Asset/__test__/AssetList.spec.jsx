import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AssetList from '../AssetList';
import AssetListItem from '../AssetListItem';

jest.mock('react-router-dom', () => ({ // eslint-disable-next-line react/prop-types
  Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

configure({ adapter: new Adapter() });

const assets = [{
  id: '1',
  name: 'Bank Card',
}, {
  id: '2',
  name: 'Cash',
}, {
  id: '3',
  name: 'Bank Card 2',
}];

describe('AssetList', () => {
  it('should render correctly', () => {
    const output = shallow(
      <AssetList assets={assets} />,
    );
    expect(output.debug()).toMatchSnapshot();
    expect(output.find(AssetListItem)).toHaveLength(assets.length);
  });
});
