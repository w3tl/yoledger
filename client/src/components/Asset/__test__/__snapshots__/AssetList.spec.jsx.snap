// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetList should render correctly 1`] = `
"<ul>
  <AssetListItem url=\\"\\" asset={{...}} />
  <AssetListItem url=\\"\\" asset={{...}} />
  <AssetListItem url=\\"\\" asset={{...}} />
</ul>"
`;
